pkgs/top-level/all-packages.nix

```diff
@@ -18084,11 +18085,23 @@ with pkgs;
     electron_29-bin
     electron_30-bin;
 
+  # Use the first package available on the current host platform
+  # and merge all meta.platforms in the resulting derivation.
+  firstAvailableOnHost = packages:
+    let platforms = lib.unique (lib.concatMap (p: p.meta.platforms) packages);
+    in let firstAvailable =
+         lib.findFirst (lib.meta.availableOn stdenv.hostPlatform)
+                       (let p = lib.head packages;
+                            name = p.pname or p.name;
+                        in (throw "${name} not supported on ${stdenv.hostPlatform.system}"))
+                       packages;
+       in lib.recursiveUpdate firstAvailable { meta.platforms = platforms; };
```

Comment:
We would also need to do something about badPlatforms.  Currently if both packages have the same badPlatforms that information would be incorrectly discarded.